### ISSUE TITLE
question marks are used instead of Russian characters in the request

### ISSUE THREAD

**Issue by djtemasuper** [PortSwigger/turbo-intruder#138]
hello, why when I use Russian characters anywhere in the script, instead of them Turbo Intruder puts question marks everywhere, moreover, when I try to encode a string with Russian characters through str.encode('utf-8'), this function does not convert the string in the Turbo Intruder script although everything works correctly in the python interpreter.
![image](https://github.com/PortSwigger/turbo-intruder/assets/142630140/4ff58ae6-5f69-40b1-a1e0-f8c53cf89c8b)
![image](https://github.com/PortSwigger/turbo-intruder/assets/142630140/582880c3-4908-476a-9641-7fcfd964b6d1)
![image](https://github.com/PortSwigger/turbo-intruder/assets/142630140/b75aa88a-27db-418c-9b52-a6d3fbae712b)

**Comment by albinowax** [comment]
Turbo Intruder uses Jython which is based on Python 2.7 which stores strings as ASCII. You're using Python 3 locally which handles strings completely differently.

**Comment by albinowax** [comment]
I'm leaving this open because it's definitely possible there's still Turbo Intruder bugs in this area. If you aren't able to fix your issue it would be useful to have a script that demonstrates the exact problem you're experiencing - eg trying to send a request containing russian characters, or viewing a response with them.

**Comment by djtemasuper** [comment]
I am experiencing the same problems with non ascii character encoding, different engines have different problems:
1) HTTP2 Engine problem with encoding in the response, instead of the response in the body of the request an incomprehensible set of characters is displayed.
Request:

```
POST /post HTTP/2
Host: httpbingo.org
Content-Length: 10
Accept: application/json
Content-Type: text/html; charset=utf-8;
Accept-Encoding: gzip, deflate, br
Accept-Language: ru-RU,ru;q=0.9,en-US;q=0.8,en;q=0.7

пы
```
Response:
![image](https://github.com/PortSwigger/turbo-intruder/assets/160190246/807e2e72-8300-48a8-a95b-178b8aa99f99)

2) BURP2 Engine due to a problem with the encoding of the request body, the server does not respond to the request (errors 400 Bad request, and as in the example, the response in general did not come).
Request:
```
POST /post HTTP/2
Host: httpbingo.org
Content-Length: 10
Accept: application/json
Content-Type: text/html; charset=utf-8;
Accept-Encoding: gzip, deflate, br
Accept-Language: ru-RU,ru;q=0.9,en-US;q=0.8,en;q=0.7

пы
```
Response:
![image](https://github.com/PortSwigger/turbo-intruder/assets/160190246/3d91c6ae-2c6f-4b07-98c2-cb99aea61f83)

**Comment by albinowax** [comment]
@djtemasuper with the HTTP2 engine, that looks like brotli encoding. Try removing the 'accept-encoding' request header.

**Comment by djtemasuper** [comment]
> @djtemasuper with the HTTP2 engine, that looks like brotli encoding. Try removing the 'accept-encoding' request header.
Yes, indeed, in the HTTP2 engine the problem was in the Accept-Encoding header, without it the response comes in normally

**Comment by Jsnsbjdjj** [comment]
Turbo Intruder modding is not modding what is the reason
